app.js

```diff
@@ -0,0 +1,200 @@
+};
+
+Bartender.prototype.createDrink = function() {
+    var tempPreferences;
+    if (isCustomer) {
+        tempPreferences = customerPreferences;
+    } else {
+        tempPreferences = preferences;
+    }
+    var drinksIngredients = [];
+    var drinkName = [];
+    for (var key in tempPreferences) {
+        if (tempPreferences[key] === "Yes") {
+            drinkName.push(key);
+            for (var i = 0; i < pantry.type.length; i++) {
+                if (pantry.type[i].category === key) {
+                    var randomizeIngredient = Math.floor((Math.random() * 3));
+                    if (randomizeIngredient && (pantry.type[i].stockValue > 0)) {
+                        drinksIngredients.push(pantry.type[i].ingredient);
+                        pantry.type[i].stockValue--;
+                        if (pantry.type[i].stockValue === 0) {
+                            restock.push({
+                                index: i,
+                                ingredient: pantry.type[i].ingredient,
+                                restock: true
+                            });
+                            displayRestockItems();
+                        }
+                        else {
+                            $(".restock").css("visibility","hidden");
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if(drinksIngredients.length > 0) {
+        restockBool = true;
+        var randomNameIndex = Math.floor((Math.random() * drinkName.length-1) + 1);
+        var randomPirateIndex = Math.floor((Math.random() * pirateDrinkNames.length-1) + 1);
+        drinks.push(new Drink(drinksIngredients, drinkName[randomNameIndex]+" "+pirateDrinkNames[randomPirateIndex]));
+    }
+    else {
+        restockBool = false;
+    }
+};
+
+var bartender = new Bartender("bartender");
+
+//Questions
+var questions = [new Question("Do ye like yer drinks strong?", "strong"),
+    new Question("Do ye like it with a salty tang?", "salty"),
+    new Question("Are ye a lubber who likes it bitter?", "bitter"),
+    new Question("Would ye like a bit of sweetness with yer poison?", "sweet"),
+    new Question("Are ye one for a fruity finish?", "fruity")
+];
+
+var ingredients = [new Ingredient("Glug of rum", "strong", 10), new Ingredient("slug of whisky", "strong", 10), new Ingredient("splash of gin", "strong", 10), new Ingredient("Olive on a stick", "salty", 10), new Ingredient("salt-dusted rim", "salty", 10), new Ingredient("rasher of bacon", "salty", 10), new Ingredient("Shake of bitters", "bitter", 10), new Ingredient("splash of tonic", "bitter", 10), new Ingredient("twist of lemon peel", "bitter", 10), new Ingredient("Sugar cube", "sweet", 10), new Ingredient("spoonful of honey", "sweet", 10), new Ingredient("splash of cola", "sweet", 10), new Ingredient("Slice of orange", "fruity", 10), new Ingredient("dash of cassis", "fruity", 10), new Ingredient("cherry on top", "fruity", 10)];
+
```

Comment:
There's a lot of declarations here, from the top down, roughly 100 lines. It could be a lot cleaner if you grouped them together. It also doesn't seem to create new objects for each question.